src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -372,21 +372,31 @@ public HttpClientSettingsKey getHttpClientKey() throws SnowflakeSQLException {
       boolean httpUseProxy = Boolean.parseBoolean(systemGetProperty("http.useProxy"));
       String httpProxyHost = systemGetProperty("http.proxyHost");
       String httpProxyPort = systemGetProperty("http.proxyPort");
+      String httpProxyUser = (systemGetProperty("http.proxyUser") != null) ? systemGetProperty("http.proxyUser") : "";
+      String httpProxyPassword = (systemGetProperty("http.proxyPassword") != null) ? systemGetProperty("http.proxyPassword") : "";
       String httpsProxyHost = systemGetProperty("https.proxyHost");
       String httpsProxyPort = systemGetProperty("https.proxyPort");
+      String httpsProxyUser = (systemGetProperty("https.proxyUser") != null) ? systemGetProperty("https.proxyUser") : "";
+      String httpsProxyPassword = (systemGetProperty("https.proxyPassword") != null) ? systemGetProperty("https.proxyPassword") : "";
       String httpProxyProtocol = systemGetProperty("http.proxyProtocol");
       String noProxy = systemGetEnv("NO_PROXY");
       String nonProxyHosts = systemGetProperty("http.nonProxyHosts");
       // log the JVM parameters that are being used
       if (httpUseProxy) {
         logger.debug(
-            "http.useProxy={}, http.proxyHost={}, http.proxyPort={}, https.proxyHost={},"
-                + " https.proxyPort={}, http.nonProxyHosts={}, NO_PROXY={}, http.proxyProtocol={}",
+            "http.useProxy={}, http.proxyHost={}, http.proxyPort={}, http.proxyUser={},"
+                + "http.proxyPassword={}, https.proxyHost={}, https.proxyPort={}, "
```

Comment:
Are you sure we want to log password? I believe it would be better just to log `*********` or empty string, WDYT?

Reply:
I have masked it explicitly.